tests/unit/factories.cpp

```diff
@@ -849,10 +848,12 @@ void grb_program( const size_t &n, RC &rc ) {
 	std::cout << "Testing matrices::ones\n";
 	test_factory_ones( rc, n );
 
-	// Check return code for distributed backends
-	ASSERT_RC_SUCCESS(
-		collectives<>::allreduce( rc, grb::operators::any_or< RC >() )
-	);
+	// synchronise return code for distributed backends
+	if( collectives<>::allreduce( rc, grb::operators::any_or< RC >() )
+		!= SUCCESS )
+	{
+		rc = PANIC;
+	}
```

Comment:
We will loose the original error by assigning PANIC, maybe a print to stderr before ?

Reply:
ok, making (and testing) the change